packages/hierarchies-react/src/presentation-hierarchies-react/itwinui/TreeErrorRenderer.tsx

```diff
@@ -79,73 +113,44 @@ export function TreeErrorRenderer({ errorList, reloadTree, scrollToElement, getH
-    />
-  );
-}
-
-type ResultSetTooLargeProps = {
-  limit: number;
-  onFilterClick?: () => void;
-  onOverrideLimit?: (limit: number) => void;
-} & LinkedNodeProps;
-
-function ResultSetTooLarge({ errorNode, onFilterClick, limit, onOverrideLimit, scrollToElement }: ResultSetTooLargeProps) {
-  const { localizedStrings } = useLocalizationContext();
-  const supportsFiltering = !!onFilterClick;
-  const supportsLimitOverride = !!onOverrideLimit && limit < MAX_LIMIT_OVERRIDE;
-  const messageWithLimit = localizedStrings.resultLimitExceeded.replace("{{limit}}", limit.toString());
+type ErrorItemContainerProps = {
+  errorNode: ErrorNode;
+  message: string;
+  actions?: { action: () => void; label: string; condition: () => boolean }[];
+} & Pick<MessageWithLinkProps, "scrollToElement">;
 
+function ErrorItemContainer({ errorNode, message, actions, scrollToElement }: ErrorItemContainerProps) {
   return (
     <ErrorRegion.Item
-      message={<MessageWithNode errorNode={errorNode} scrollToElement={scrollToElement} message={messageWithLimit} />}
+      message={<MessageWithLink linkLabel={errorNode.parent?.label} scrollToElement={scrollToElement} message={message} />}
       actions={
         <div style={{ display: "flex", flexDirection: "column" }}>
-          {supportsLimitOverride && (
-            <Anchor onClick={() => onOverrideLimit(MAX_LIMIT_OVERRIDE)}>
-              {localizedStrings.increaseHierarchyLimit.replace("{{limit}}", MAX_LIMIT_OVERRIDE.toString())}
-            </Anchor>
+          {actions?.map(
+            (actionItem) =>
+              actionItem.condition() && (
+                <Anchor key={actionItem.label} onClick={actionItem.action}>
+                  {actionItem.label}
+                </Anchor>
+              ),
           )}
-          {supportsFiltering && <Anchor onClick={onFilterClick}>{localizedStrings.increaseHierarchyLimitWithFiltering}</Anchor>}
         </div>
       }
     />
   );
 }
 
-/** @alpha */
-interface LinkedNodeProps {
-  errorNode: ErrorNode;
-  scrollToElement: (node: ErrorNode) => void;
-}
-
-function LinkedNode({ errorNode, scrollToElement }: LinkedNodeProps) {
-  return <Anchor onClick={() => scrollToElement(errorNode)}>{errorNode.parent?.label}</Anchor>;
+interface MessageWithLinkProps {
+  scrollToElement: () => void;
+  message: string;
+  linkLabel?: string;
 }
 
-function MessageWithNode({ errorNode, scrollToElement, message }: LinkedNodeProps & { message: string }) {
+function MessageWithLink({ linkLabel, scrollToElement, message }: MessageWithLinkProps) {
   const splitMessage = message.split("{{node}}", 2);
   return (
     <div style={{ display: "flex", whiteSpace: "pre", flexWrap: "wrap" }}>
       <Text variant={"body-sm"}>{splitMessage[0]}</Text>
-      <LinkedNode errorNode={errorNode} scrollToElement={scrollToElement} />
+      <Anchor onClick={scrollToElement}>{linkLabel}</Anchor>
```

Comment:
should we render this if `linkLabel` is `undefined`?

Reply:
We don't really expect it to be undefined, but with current setup Root Node's parent is undefined, but I don't think it's possible to have rootNode be an errorItem.

Reply:
> I don't think it's possible to have rootNode be an errorItem.

wouldn't that be the case if we fail to load the root level (e.g. too many root nodes)?

Reply:
Started working on a root error renderer, seems like it would be more suited for a separate issue/PR, we'd need to figure out how we want to handle root node filtering as we currently don't have an ability to do so.

Reply:
Please file an issue and link it here.

Reply:
https://github.com/iTwin/viewer-components-react/issues/1290